Flaky `TestOverridesExporterRing_scaleDown`

The test `TestOverridesExporterRing_scaleDown` is flaky, see this build https://github.com/grafana/mimir/actions/runs/3948592087/jobs/6758766697

```
--- FAIL: TestOverridesExporterRing_scaleDown (5.02s)
    ring_test.go:102: expected LEAVING, got ACTIVE
FAIL
FAIL	github.com/grafana/mimir/pkg/util/validation/exporter	6.103s
```

--- a/pkg/util/validation/exporter/ring_test.go
+++ b/pkg/util/validation/exporter/ring_test.go
@@ -100,7 +100,7 @@ func TestOverridesExporterRing_scaleDown(t *testing.T) {
 
 	// Wait for the leader to have advertised its leaving state to the ring
 	test.Poll(t, 5*time.Second, ring.LEAVING, func() interface{} {
-		rs, _ := i1.client.GetAllHealthy(ringOp)
+		rs, _ := i2.client.GetAllHealthy(ringOp)
 		for _, instance := range rs.Instances {
 			if instance.Addr == l1.GetInstanceAddr() {
 				return instance.GetState()

